core/src/mender-artifact.c

```diff
@@ -392,7 +394,15 @@ mender_artifact_parse_tar_header(mender_artifact_ctx_t *ctx) {
     ctx->file.name = tmp;
 
     /* Retrieve file size */
-    sscanf(tar_header->size, "%o", (unsigned int *)&(ctx->file.size));
+    char *end_ptr;
+    errno          = 0; /* to distinguish between success/failure */
+    ctx->file.size = strtoul(tar_header->size, &end_ptr, 8);
```

Comment:
In theory, the size is up to 11 digits (see tar format [here](https://www.gnu.org/software/tar/manual/html_node/Standard.html), the array is 12 but the last byte will be `\0`). With `unsigned long` is enough, right? Or should we use `long long` to be safe?

(Honest question)

Reply:
A long can be 4 or 8 Bytes. In case of a 4 Byte unsigned long the largest value it can hold would be
```python3
>>> pow(2, 32) - 1
4294967295
```

Reply:
The largest number 11 digits in base 8 can represent is
```python3
>>> sum = 0
>>> for i in range(10, -1, -1):
...     sum += (7 * 8**i)
... 
>>> sum
8589934591
```

Reply:
So I guess we should can use `strtoull(3)` instead. Nice catch 🚀